From: David Heidelberg <david@ixit.cz>
Subject: mm/vmalloc: warn only once when vmalloc detect invalid gfp flags
Date: Tue, 18 Nov 2025 01:05:57 +0100

Without WARN_ONCE, the logs get spammed immediately after the boot, on
devices as OnePlus 6T (Snapdragon 845).

Link: https://lkml.kernel.org/r/20251118-only-one-vmalloc-v2-1-fe702aa68d8b@ixit.cz
Link: https://lkml.kernel.org/r/20251118-only-one-vmalloc-v1-1-29f8eacb4605@ixit.cz
Fixes: 7179b2256315 ("mm/vmalloc: warn on invalid vmalloc gfp flags")
Signed-off-by: David Heidelberg <david@ixit.cz>
Reviewed-by: Vishal Moola (Oracle) <vishal.moola@gmail.com>
Reviewed-by: Uladzislau Rezki (Sony) <urezki@gmail.com>
Cc: Peng Li <peng8420.li@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/vmalloc.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/vmalloc.c~mm-vmalloc-warn-only-once-when-vmalloc-detect-invalid-gfp-flags
+++ a/mm/vmalloc.c
@@ -3936,8 +3936,8 @@ static gfp_t vmalloc_fix_flags(gfp_t fla
 	gfp_t invalid_mask = flags & ~GFP_VMALLOC_SUPPORTED;
 
 	flags &= GFP_VMALLOC_SUPPORTED;
-	WARN(1, "Unexpected gfp: %#x (%pGg). Fixing up to gfp: %#x (%pGg). Fix your code!\n",
-			invalid_mask, &invalid_mask, flags, &flags);
+	WARN_ONCE(1, "Unexpected gfp: %#x (%pGg). Fixing up to gfp: %#x (%pGg). Fix your code!\n"
+		  invalid_mask, &invalid_mask, flags, &flags);
 	return flags;
 }
 
_
